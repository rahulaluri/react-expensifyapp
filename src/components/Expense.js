import React from 'react';
import {Link} from 'react-router-dom';
import numeral from 'numeral';
import moment from 'moment';


const ExpenseListItem = ({dispatch, id,description, amount, createdAt}) => (
    <div>
       <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
        <p>{numeral(amount).format('$0,0.00')}
            -
        {moment(createdAt).format("MMM Do, YYYY")}</p>    
    </div>
)

export default ExpenseListItem;

