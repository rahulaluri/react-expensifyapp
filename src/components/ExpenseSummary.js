import React from 'react';
import numeral from 'numeral';
import {connect} from 'react-redux';
import ExpenseListFilters from '../selectors/expenses';
import SumOfExpenses from '../selectors/Expenses-total';

const ExpensesSummary = ({expensesCount, expensesTotal}) =>{
    const expenseWord = expensesCount===1 ? 'expense' : 'expenses';
    const formatterExpenseTotal = numeral(expensesTotal).format('$0,0.00');
    return (
        <div>
        <h1>expenses total for {expensesCount}{expenseWord} is {formatterExpenseTotal} </h1>
        </div>
    )
}

const mapStateToProps =(state) => {
     
        const visibleExpenses = ExpenseListFilters(state.expenses,state.filters)
        return {
            expensesTotal: SumOfExpenses(visibleExpenses),
            expensesCount: visibleExpenses.length 
        }
       
 
    
    }

export default connect(mapStateToProps)(ExpensesSummary);
