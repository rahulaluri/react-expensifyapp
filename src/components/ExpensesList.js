import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './Expense';
import selectExpenses from '../selectors/expenses';
import ExpenseListFilters from './ExpenseListFilters';

export const ExpenseList = (props) => (
    <div>
        <ExpenseListFilters />
        {props.expenses.length===0 ? <p>AddExpenses</p> : (props.expenses.map((expense)=><ExpenseListItem key={expense.id} {...expense}/> ))}       
    </div>
);


const mapStatetoProps = (state) =>{
    return  {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStatetoProps)(ExpenseList)




