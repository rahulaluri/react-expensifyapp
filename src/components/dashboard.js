import React from 'react';
import ExpenseList from './ExpensesList';
const ExpenseDashboard = () => (
  <div>
    This is from my dashboard component!
    <ExpenseList />
  </div>
);

export default ExpenseDashboard;
