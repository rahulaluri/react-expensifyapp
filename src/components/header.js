import React from 'react';
import {BrowserRouter,Route,Switch,Link,NavLink} from 'react-router-dom';

const Header = ()=> (
    <header>
    <h1> Expensify</h1>
   <NavLink activeClassName="is-active" to="/" exact={true}>Dashboard </NavLink> 
   <NavLink activeClassName="is-active" to="/create">Create </NavLink>
   </header>
);

export default Header;