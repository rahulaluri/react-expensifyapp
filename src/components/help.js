import React from 'react';

const HelpPage = ()=>(
    <div>
    You have hit the help page
    </div>
    );

export default HelpPage;

