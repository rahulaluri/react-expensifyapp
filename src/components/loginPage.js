import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

const Login = ({startLogin}) => (
    <div>
        <h1>Login Here</h1>
        <button onClick={startLogin}>Login</button>
    </div>
)

const mapDispatchToProps= (dispatch)=>({
startLogin: ()=> dispatch(startLogin())
})
export default connect(undefined,mapDispatchToProps)(Login);