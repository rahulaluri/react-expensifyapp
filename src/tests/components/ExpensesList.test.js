import React from 'react';
import {shallow} from 'enzyme'
import {ExpenseList} from '../../components/ExpensesList';

const expenses = [{
    description:'love',
    note: '12312',
    amount: 0,
    createdAt:0
}, {description:'loadave',
note: '123asd12',
amount: 0,
createdAt:0}];

test('Should render the expense list correctly', ()=>{
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
})

test('Should render the empty expense list correctly', ()=>{
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
})

