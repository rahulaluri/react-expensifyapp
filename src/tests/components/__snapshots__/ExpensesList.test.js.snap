// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the empty expense list correctly 1`] = `
<div>
  <Connect(ExpenseListFilters) />
  <p>
    AddExpenses
  </p>
</div>
`;

exports[`Should render the expense list correctly 1`] = `
<div>
  <Connect(ExpenseListFilters) />
  <ExpenseListItem
    amount={0}
    createdAt={0}
    description="love"
    note="12312"
  />
  <ExpenseListItem
    amount={0}
    createdAt={0}
    description="loadave"
    note="123asd12"
  />
</div>
`;
